### PR TITLE
chore(rust-client): Update to latest master

### PR DESCRIPTION
**Pull request by mxinden**

### CODE REVIEW

**Comment by dignifiedquire** [comment]
CI seems sad

**Comment by mxinden** [comment]
Failure is in the NixOS build. @MarcoPolo would you mind taking a look?

```
Run nix build .#rust-client
copying path '/nix/store/26mbv2f99wvagc75wwa7ai1di7x9l7jl-source' from 'https://cache.nixos.org/'...
error: A hash was specified for libp2p-0.50.0, but there is no corresponding git dependency.
(use '--show-trace' to show detailed location information)
Error: Process completed with exit code 1.
```

I am not sure to what values to update the below.

https://github.com/libp2p/punchr/blob/6a99189cea8c91205c12f4ad6c4731ff2c812243/flake.nix#L68-L71

**Comment by MarcoPolo** [comment]
Fixed. You changed the dep from 0.50 to 0.51, so you had to update the flake.nix to specify the expected hash of 0.51. Nix needs to know what the checksum of the Git dependency since it's not included in the Cargo.lock file.

File: flake.nix [hunk 1]
```diff
@@ -67,8 +67,8 @@
             outputHashes = {
               # Update the hash by using a fake hash, and then updating it with
               # the correct one when the build fails
-              # "libp2p-0.50.0" = pkgs.lib.fakeSha256;
-              "libp2p-0.50.0" = "sha256-9l3i5UwgJGihDfw8Q+zf5nd0FjjAOhK/nyhnqWO18Ko=";
+              # "libp2p-0.51.0" = pkgs.lib.fakeSha256;
+              "libp2p-0.51.0" = "sha256-pHmud2Km+zL34hUN25vldI9oMJnP47cnessZrzt43BE=";
```
Comment (mxinden) [inline flake.nix:71]: What is the preimage of this hash @MarcoPolo? I fail to identify this as a git hash.

**Pull request merged**